### ISSUE TITLE
Add-on store attempts to validate file internal to add-on.

### ISSUE THREAD

**Issue by fredrikbaberg** [home-assistant/supervisor#2754]
### Describe the issue you are experiencing

The supervisor (Add-on store?) attempts to validate a config file internal to add-on. File is in a subfolder of the add-on, irrelevant for presentation in the Add-on store.

### What is the used version of the Supervisor?

supervisor-2021.03.6

### What type of installation are you running?

Home Assistant OS

### Which operating system are you running on?

Home Assistant Operating System

### What is the version of your installed operating system?

5.12

### What version of Home Assistant Core is installed?

core-2021.3.4

### Steps to reproduce the issue

1. Add repository `https://github.com/fredrikbaberg/hassio-addons` to Add-on store.
2. Reload.
3. See warning in supervisor log.
...

### Anything in the Supervisor logs that might be useful for us?

```txt
# Put your logs below this line

21-03-26 08:53:26 WARNING (MainThread) [supervisor.store.data] Can't read /data/addons/git/774437fd/octoprint_dev/FS/octoprint/config.yaml: Invalid Add-on config!. Got ordereddict([('accessControl', ordereddict([('autologinAs', 'homeassistant'), ('autologinLocal', True), ('localNetworks', ['172.30.32.2'])])), ('folder', ordereddict([('data', '/data/octoprint/data'), ('generated', '/data/octoprint/generated'), ('logs', '/data/octoprint/logs'), ('plugins', '/data/octoprint/plugins'), ('virtualSd', '/data/octoprint/virtualSd'), ('timelapse_tmp', '/data/octoprint/timelapse/tmp'), ('printerProfiles', '/config/octoprint/printerProfiles'), ('slicingProfiles', '/co...
21-03-26 08:53:27 WARNING (MainThread) [supervisor.store.data] Can't read /data/addons/git/774437fd/octoprint/FS/octoprint/config.yaml: Invalid Add-on config!. Got ordereddict([('accessControl', ordereddict([('autologinAs', 'homeassistant'), ('autologinLocal', True), ('localNetworks', ['172.30.32.2'])])), ('folder', ordereddict([('data', '/data/octoprint/data'), ('generated', '/data/octoprint/generated'), ('logs', '/data/octoprint/logs'), ('plugins', '/data/octoprint/plugins'), ('virtualSd', '/data/octoprint/virtualSd'), ('timelapse_tmp', '/data/octoprint/timelapse/tmp'), ('printerProfiles', '/config/octoprint/printerProfiles'), ('slicingProfiles', '/co...
21-03-26 08:53:27 WARNING (MainThread) [supervisor.store.data] Can't read /data/addons/git/774437fd/octoprint_dev/FS/octoprint/config.yaml: Invalid Add-on config!. Got ordereddict([('accessControl', ordereddict([('autologinAs', 'homeassistant'), ('autologinLocal', True), ('localNetworks', ['172.30.32.2'])])), ('folder', ordereddict([('data', '/data/octoprint/data'), ('generated', '/data/octoprint/generated'), ('logs', '/data/octoprint/logs'), ('plugins', '/data/octoprint/plugins'), ('virtualSd', '/data/octoprint/virtualSd'), ('timelapse_tmp', '/data/octoprint/timelapse/tmp'), ('printerProfiles', '/config/octoprint/printerProfiles'), ('slicingProfiles', '/co...
21-03-26 08:53:27 WARNING (MainThread) [supervisor.store.data] Can't read /data/addons/git/774437fd/octoprint/FS/octoprint/config.yaml: Invalid Add-on config!. Got ordereddict([('accessControl', ordereddict([('autologinAs', 'homeassistant'), ('autologinLocal', True), ('localNetworks', ['172.30.32.2'])])), ('folder', ordereddict([('data', '/data/octoprint/data'), ('generated', '/data/octoprint/generated'), ('logs', '/data/octoprint/logs'), ('plugins', '/data/octoprint/plugins'), ('virtualSd', '/data/octoprint/virtualSd'), ('timelapse_tmp', '/data/octoprint/timelapse/tmp'), ('printerProfiles', '/config/octoprint/printerProfiles'), ('slicingProfiles', '/co...

```

Bug is not breaking, the Add-on still install and runs, but the behaviour and warning is unexpected.

This was reported on the add-on GitHub (https://github.com/fredrikbaberg/hassio-addons/issues/123). I've verified that it occurs on my Raspberry Pi 3 install (used to fill in this form) and on a VM.

I'm the author of the add-on that reports this issue. If this is purely due to how I've created the Add-on, please let me know solution or where to ask for more help.

**Comment by ludeeus** [comment]
The Supervisor can't know where in your repository the correct add-on config file is located, so it does a recursive lookup for any file matching config.(yml|yaml|json), if you rename your other configfile from `config.yaml`  to `configuration.yaml` it will no longer be detected as the add-on configuration

**Comment by fredrikbaberg** [comment]
> The Supervisor can't know where in your repository the correct add-on config file is located, so it does a recursive lookup for any file matching config.(yml|yaml|json), if you rename your other configfile from `config.yaml` to `configuration.yaml` (or any other name) it will no longer be detected as the add-on configuration

Thank you, I'll look at renaming the file(s) in my repository to remove the warning. I've also made a PR to documentation, to help others avoid the same issue.

It does seem like a strange/unnecessary restriction on filenames, it is not an uncommon name to use?
I'm not sure if this is something I've interpreted myself or explicitly read in the documentation, but my understanding is that the structure of the repository should be `<repository root>`/`<addon folder>`.
- If this is correct, is a recursive lookup with no maximum depth really needed?
- Could the first match be used? I haven't looked at the code that does the recursive lookup yet, so not sure if it is feasible, but I'll try to look at it either this weekend or next.
